resources/js/components/ui/calendar/ResourcesSelector.tsx

```diff
@@ -0,0 +1,130 @@
+import {cx, htmlAttributes} from "components/utils";
+import {For, JSX, Show, VoidComponent, createMemo, createRenderEffect, createSignal, splitProps} from "solid-js";
+
+export type ResourcesSelectionMode = "radio" | "checkbox";
+
+interface Props extends htmlAttributes.div {
+  resourceGroups?: readonly ResourceGroup[];
+  mode: ResourcesSelectionMode;
+  selection: readonly string[];
+  setSelection: (ids: string[]) => void;
+}
+
+export interface ResourceGroup {
+  readonly label: () => JSX.Element;
+  readonly resources: readonly Resource[];
+}
+
+export interface Resource {
+  readonly id: string;
+  readonly label: () => JSX.Element;
+}
+
+/**
+ * A vertical list of people and resources, with checkboxes or radio buttons, depending on mode.
+ */
+export const ResourcesSelector: VoidComponent<Props> = (allProps) => {
+  const [props, divProps] = splitProps(allProps, ["resourceGroups", "mode", "selection", "setSelection"]);
+  const selected = createMemo(() => new Set(props.selection));
+  let div: HTMLDivElement | undefined;
+  return (
+    <div ref={div} {...htmlAttributes.merge(divProps, {class: "flex flex-col gap-2"})}>
+      <For each={props.resourceGroups}>
+        {({label, resources}, i) => {
+          const inputId = () => (props.mode === "checkbox" ? `resourceGroup_${i()}` : undefined);
+          const [input, setInput] = createSignal<HTMLInputElement>();
+          const state = createMemo(() => {
+            if (!inputId()) {
+              return undefined;
+            }
+            const sel = new Set(props.selection);
+            let all = true;
+            let none = true;
+            for (const resource of resources)
+              if (sel.has(resource.id)) {
+                none = false;
+              } else {
+                all = false;
+              }
+            return {all, none};
+          });
+          // Make the group checkbox indeterminate if some resources are selected and some are not.
+          createRenderEffect(() => {
+            const inp = input();
+            if (inp) {
+              const s = state();
+              inp.indeterminate = !!s && s.all === s.none;
```

Comment:
czy dopuszczasz puste resourceGroups? bo wtedy all === none

Reply:
bardziej naturalny warunek to może some(r => sel.has(r.id)) && some(r => !sel.has(r.id)), tak samo napisałeś słownie w komentarzu

Reply:
!all && !none też dobre :)

Reply:
Myślę że all===none to dobry warunek, dokładnie zresztą ogarnia to o co chodzi. Dla pustego z jakiegoś powodu mimo że sygnał dalej kazał mu nie być checked to się robił checked, chwilę podebugowałem poczym dodałem disabled i tyle.

Reply:
No mi chodziło o to że pusty nie wydaje się indeterminate tak po ludzku. Ale pusty to i tak dziwny bo rozumiem że nic z nim nie można robić. To może go całkiem pomijaj?

Reply:
Całkiem go nie chcę pomijać. Ale coś tam jeszcze zmieniłem.